lib/job-iteration/integrations.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module JobIteration
+  # @api private
+  module Integrations
+    LoadError = Class.new(StandardError)
+
+    extend self
+
+    attr_accessor :registered_integrations
+
+    self.registered_integrations = {}
+
+    autoload :Sidekiq, "job-iteration/integrations/sidekiq"
+    autoload :Resque, "job-iteration/integrations/resque"
+
+    # @api public
+    def register(name, callable)
+      raise ArgumentError, "Interruption adapter must respond to #call" unless callable.respond_to?(:call)
+
+      registered_integrations[name] = callable
+    end
+
+    def load(name)
+      if (callable = registered_integrations[name])
+        callable
+      else
+        begin
+          klass = "#{self}::#{name.camelize}".constantize
+          register(name, klass)
+        rescue NameError
+          raise LoadError, "Could not find integration for '#{name}'"
```

Comment:
This should register a dummy handler, not raise.
Otherwise, this results in an exception for any queue adapter that isn't recognized

Reply:
But this breaks the users (our team uses active_qless (yes, I know, it's unmaintained, we're moving away from it))

How about this:
- Add a graceful fallback, preserving the current behaviour, but with a deprecated notice
- Release a patch version on the current minor version
- Plan to remove the graceful fallback in the next major version

Reply:
Alright then 👍🏼